src/lib.rs

```diff
@@ -987,6 +999,23 @@ impl<A: Array> Clone for ArrayVec<A>
     }
 }
 
+#[cfg(feature="quickcheck-1")]
+impl<A: Array + Send + Clone + 'static > Arbitrary for ArrayVec<A>
+where <A as array::Array>::Item: std::clone::Clone + Arbitrary,
+<A as array::Array>::Index: std::marker::Send
+{
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        use std::cmp::{min, max};
+        let mut v = Self::default();
+        let size = g.gen_range(0, max(1, min(g.size(), v.capacity())));
```

Comment:
```suggestion
        let size = g.gen_range(0, max(1, min(g.size(), v.capacity() + 1)));
```

Since `gen_range` is exclusive on the right side, `v.capacity() + 1` should be good as well.